src/main/java/org/swisspush/redisques/util/QueueStatisticsCollector.java

```diff
@@ -0,0 +1,363 @@
+        //       order as the requested queues.
+        List<String> queueKeys = new ArrayList<>();
+        for (String queue : queues){
+            queueKeys.add(this.queuePrefix + queue);
+        }
+        luaScriptManager.handleMultiListLength(queueKeys, queueListLength -> {
+            if (queueListLength == null) {
+                log.error("Unexepected queue MultiListLength result null");
+                event.reply(new JsonObject().put(STATUS, ERROR));
+                return;
+            }
+            if (queueListLength.size() != queues.size()) {
+                log.error("Unexpected queue MultiListLength result with unequal size {} : {}",
+                    queues.size(), queueListLength.size());
+                event.reply(new JsonObject().put(STATUS, ERROR));
+                return;
+            }
+            // populate the list of queue statistics in a Hashmap for later fast merging
+            final HashMap<String, QueueStatistic> statisticsMap = new HashMap<>();
+            for (int i = 0; i < queues.size(); i++) {
+                QueueStatistic qs = new QueueStatistic(queues.get(i));
+                qs.setSize(queueListLength.get(i));
+                statisticsMap.put(qs.queueName, qs);
+            }
+            // now retrieve all available statistics from Redis and merge them
+            // together with the previous populated queue statistics map
+            redisAPI.hvals(STATSKEY, statisticsSet -> {
+                if (statisticsSet == null) {
+                    log.error("Unexepected statistics queue evaluation result result null");
+                    event.reply(new JsonObject().put(STATUS, ERROR));
+                    return;
+                }
+                // put the received statistics data to the former prepared statistics objects
+                // per queue
+                Iterator<Response> itr = statisticsSet.result().stream().iterator();
+                while (itr.hasNext()){
+                    JsonObject jObj = new JsonObject(itr.next().toString());
+                    String queueName = jObj.getString(RedisquesAPI.QUEUENAME);
+                    QueueStatistic queueStatistic = statisticsMap.get(queueName);
+                    if (queueStatistic != null) {
+                        // if it isn't there, there was obviously no statistic available for this
+                        queueStatistic.setFailures(jObj.getLong(QUEUE_FAILURES, 0L));
+                        queueStatistic.setBackpressureTime(jObj.getLong(QUEUE_BACKPRESSURE, 0L));
+                        queueStatistic.setSlowdownTime(jObj.getLong(QUEUE_SLOWDOWNTIME, 0L));
+                    }
+                }
+                // build the final resulting statistics list from the former merged queue
+                // values from various sources
+                JsonArray result = new JsonArray();
+                for (int i = 0; i < queues.size(); i++) {
+                    String queueName = queues.get(i);
```

Comment:
```java
for (int i = 0; i < queues.size(); i++) {
    String queueName = queues.get(i);
    // ...
```

Why not as [enhanced for loop](https://www.tutorialspoint.com/what-is-enhanced-for-loop-in-Java)? Eg:

```java
for (String queueName : queues) {
    // ...
```